### DISCUSSION TITLE
Tower Defense - Metrics

### DISCUSSION THREAD

**Discussion by TimWaeckerle** [Freaky2164/TowerDefense#46] [General]
Hi all!

We talked about Cohesionin the last lecture and wanted to share our slides [here](https://github.com/argastle/TowerDefense/blob/main/Projektmanagement/Coherent/Coherent.pdf).

Cohesion and Coupling are two metrics that are closely related and both have a huge impact on the quality of the architecture of software.
When it comes to software engineering the best solution is generally to keep the cohesion inside modules as high as possible while keeping the coupling between different modules as low as possible.

This leads to a more qualitative software product because it is easier to understand, to test, to modify and  to reuse. 

Because of our presentation our team is quite experienced about Cohesion and Coupling. Therefor we decided to measure those two metrics for our project. For the third metric we decided it would be interesting to measure the source code of our project. 
Because most of our team members dont work with C++ or Unity we are interested to see how much source code we actually wrote during our development.

For the calculation of the metrics we want to look into SonarQube for analyzing our source code. It was mentioned in nearly every presentation in our last lecture, so we are hopeful that it can give as a lot of insight into our project.

Greetings TD Team

**Comment by KevStr06** [comment]
Hello TD Team,

Thank you for sharing your slides on Cohesion and Coupling. It's great to see that your team is taking a proactive approach to measuring these metrics for your project. By striving to maintain high cohesion and low coupling, you are on the right track to producing a higher quality software product that is easier to understand, test, modify, and reuse.

Using SonarQube to analyze your source code is a great idea, as it can provide valuable insights into the overall quality of your codebase. Keep in mind, however, that SonarQube is just one tool and should be used in conjunction with other metrics and best practices to ensure that your code is of the highest quality.

Good luck with your project and keep up the good work!

Team Cybuy

**Comment by Mohmad-Naser-alnakeshbandi** [comment]
Hi TD Team, 

Great presentation on cohesion and coupling! It's evident that you have a strong understanding of these important software architecture metrics. Keeping high cohesion within modules while minimizing coupling between different modules is indeed crucial for developing a high-quality software product.

I appreciate your team's decision to measure cohesion and coupling metrics for your project. By doing so, you'll gain valuable insights into the overall architecture and improve the software's understandability, testability, modifiability, and reusability.

It's also interesting that you're planning to measure the size of your source code as a third metric. This can provide additional information about the scope and complexity of your project.

Using SonarQube for analyzing your source code is an excellent choice. It's a powerful tool widely recommended in the software engineering community for code quality analysis. I'm confident that it will provide you with valuable insights and help you make informed decisions to enhance your project.

Best of luck with your measurements and analysis! I'm sure the results will provide you with valuable information to improve your software development process.

Best Regards, 
Ghalia Team
